### ISSUE TITLE
Setup Railway for Continuous Deployment

### ISSUE THREAD

**Issue by 0x4007** [pavlovcik/uad-ui#443]
This looks like another good and free solution for deployments. Ideally if somebody has experience setting up Railway this should be a pretty easy task to complete.

1. Fork repo
2. Setup Railway on your fork
3. Link proof that it works, and work with me to share config so that I can set it up on our end.

https://railway.app/

**Comment by sprtd** [comment]
I'm interested in this

**Comment by sergfeldman** [comment]
> I'm interested in this

@sprtd Hey, do you have questions on this issue or everything is clear?

**Comment by sprtd** [comment]
> > I'm interested in this
> 
> @sprtd Hey, do you have questions on this issue or everything is clear?

@sergfeldman Please could you clarify if the CD is meant  to track both packages - `contracts` and `dapp` folders or it's meant  for the `dapp` folder only

**Comment by 0x4007** [comment]
> > > I'm interested in this
> > 
> > 
> > @sprtd Hey, do you have questions on this issue or everything is clear?
> 
> @sergfeldman Please could you clarify if the CD is meant to track both packages - `contracts` and `dapp` folders or it's meant for the `dapp` folder only

Taking a look at our [Netlify continuous deployment script](https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/install-and-build-pr.yml) may provide some additional context. From what I understand, we compile the smart contracts and render the ABIs. Those ABIs are used by the front end which is in the `dapp` folder.

**Comment by FibrinLab** [comment]
> > > > I'm interested in this
> 
> > > 
> 
> > > 
> 
> > > @sprtd Hey, do you have questions on this issue or everything is clear?
> 
> > 
> 
> > @sergfeldman Please could you clarify if the CD is meant to track both packages - `contracts` and `dapp` folders or it's meant for the `dapp` folder only
> 
> 
> 
> Taking a look at our [Netlify continuous deployment script](https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/install-and-build-pr.yml) may provide some additional context. From what I understand, we compile the smart contracts and render the ABIs. Those ABIs are used by the front end which is in the `dapp` folder.

I set this up for my own dapp. Let me setup using the fork on my account. I will share confit steps

**Comment by FibrinLab** [comment]
> > > > I'm interested in this
> > > 
> > > 
> > > @sprtd Hey, do you have questions on this issue or everything is clear?
> > 
> > 
> > @sergfeldman Please could you clarify if the CD is meant to track both packages - `contracts` and `dapp` folders or it's meant for the `dapp` folder only
> 
> Taking a look at our [Netlify continuous deployment script](https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/install-and-build-pr.yml) may provide some additional context. From what I understand, we compile the smart contracts and render the ABIs. Those ABIs are used by the front end which is in the `dapp` folder.

@pavlovcik Railway can be setup unless the next js build errors are fixed

**Comment by 0x4007** [comment]
You could rollback to a functional commit, like what we're using on uad.ubq.fi its pointing to this PR https://github.com/ubiquity/ubiquity-dollar/pull/408

**Comment by 0x4007** [comment]
@Steveantor will mark as complete after you verify its working

**Comment by rndquu** [comment]
@Steveantor @pavlovcik 

Railway deploy code is commented right now https://github.com/ubiquity/ubiquity-dollar/blob/development/.github/workflows/continuous-deploy.yml#L52

Do we really need the railway deploy script? We already deploy our frontend to netlify and cloudflare pages. Adding railway seems to be redundant.

**Comment by 0x4007** [comment]
Thanks for the thoughtful question @rndquu 

I believe I commented it out because it wasn't working when I tried using it.

I like the idea of setting up redundant CI/CD because in the past we have seen some providers have infrastructure issues which caused our deployments to fail (Fleek, which has been totally removed from our code at this point.) Redundancy with various providers makes our UI hosting, and CI/CD more resilient. It also reminds me of when Cloudflare introduced the requirement of adding the `.nvmrc` file with the node version seemingly without warning. Fortunately we were able to continue hosting the UI with our other provider (at the time it might have been Fleek or Vercel.) which helps maintain our development velocity.

Beyond just for development purposes, ideologically in order to increase our protocol's censorship resistance, redundancy is great to host on separate providers. I am onboard with using as many as we can find as long as they are free of charge!
